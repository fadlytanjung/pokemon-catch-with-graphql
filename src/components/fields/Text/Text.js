import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Text(props) {
  const { className, input, inputProps, message, meta } = props;
  const { dirty, error, touched } = meta;
  const classes = [
    styles.root,
    error &&  (dirty || touched) && styles.error,
    className,
  ].filter(Boolean).join(' ');
  const validation = (typeof error === 'string') && error;

  return (
    <>
      {(inputProps.type !== 'hidden') && <label className={styles.label}>{inputProps.label}</label>}
      <input className={classes} {...input} {...inputProps} />
      {error &&  (dirty || touched) && <small className={styles.error}>{validation || message}</small>}
    </>
  );
}

Text.defaultProps = {
  className: '',
  input: {},
  inputProps: {},
  message: '',
  meta: {},
};

Text.propTypes = {
  className: PropTypes.string,
  input: PropTypes.object,
  inputProps: PropTypes.object,
  message: PropTypes.string,
  meta: PropTypes.object,
};
