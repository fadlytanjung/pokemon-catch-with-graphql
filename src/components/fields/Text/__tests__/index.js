import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Text from '../Text';

describe('src/components/fields/Text', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Text />);
    expect(tree).toMatchSnapshot();

    const result = Text({
      ...Text.defaultProps,
      input: { value: 'test' },
      inputProps: { label: 'abc' },
    });
    const label = result.props.children[0].props.children;
    expect(label).toBe('abc');
  });

  test('render error message', () => {
    const meta = { error: true, touched: true };
    const message = 'message';
    const res = Text({ ...Text.defaultProps, message, meta });
    expect(res.props.children[2].props.children).toBe('message');
  });

  test('render error validation', () => {
    const meta = { error: 'error', touched: true };
    const res = Text({ ...Text.defaultProps, meta });
    expect(res.props.children[2].props.children).toBe('error');
  });
});
