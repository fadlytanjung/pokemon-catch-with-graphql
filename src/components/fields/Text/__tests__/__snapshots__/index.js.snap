// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Text render 1`] = `
<React.Fragment>
  <label />
  <input
    className=""
  />
</React.Fragment>
`;
